### PR TITLE
Move all tests over to Rely

### PR DESCRIPTION
**Pull request by ulrikstrid**
This allows us to test on Windows and we also get nice jUnit reporting

### CODE REVIEW

File: azure-pipelines.yml [hunk 1]
```diff
@@ -6,9 +6,9 @@ stages:
   - job:
     strategy:
       matrix: {
-        # windows_406: { OCAML_VERSION: "4.06", platform: Windows, vmImage: windows-latest, CACHE_FOLDER: $(Pipeline.Workspace)\cache },
-        # windows_407: { OCAML_VERSION: "4.07", platform: Windows, vmImage: windows-latest, CACHE_FOLDER: $(Pipeline.Workspace)\cache },
-        # windows_408: { OCAML_VERSION: "4.08", platform: Windows, vmImage: windows-latest, CACHE_FOLDER: $(Pipeline.Workspace)\cache },
+        windows_406: { OCAML_VERSION: "4.06", platform: Windows, vmImage: windows-latest, CACHE_FOLDER: $(Pipeline.Workspace)\cache },
+        windows_407: { OCAML_VERSION: "4.07", platform: Windows, vmImage: windows-latest, CACHE_FOLDER: $(Pipeline.Workspace)\cache },
+        windows_408: { OCAML_VERSION: "4.08", platform: Windows, vmImage: windows-latest, CACHE_FOLDER: $(Pipeline.Workspace)\cache },
```
Comment (Et7f3) [inline azure-pipelines.yml]: :heart_eyes: we can dev it on windows.

File: esy.lock/index.json [hunk 2]
```diff
@@ -1836,34 +830,6 @@
...
-          "opamoverride":
-            "esy.lock/overrides/opam__s__num_opam__c__1.2_opam_override"
-        }
-      ],
-      "dependencies": [
-        "ocaml@4.8.1000@d41d8cd9", "@opam/ocamlfind@opam:1.8.1@ff07b0f9",
-        "@esy-ocaml/substs@0.0.1@d41d8cd9"
-      ],
-      "devDependencies": [ "ocaml@4.8.1000@d41d8cd9" ]
-    },
```
Comment (Et7f3) [inline esy.lock/index.json]: So much deps removed maybe we can go below < 400 MO of cache now.

File: package.json [hunk 3]
```diff
@@ -17,12 +21,11 @@
   },
   "devDependencies": {
     "ocaml": "~4.8.0",
-    "@opam/alcotest": "^0.8.4",
     "@opam/easy-format": "^1.3.1",
```
Comment (Et7f3) [inline package.json]: Is it still needed ?

File: test/Test.re [hunk 4]
```diff
@@ -885,48 +1041,38 @@ let core = [
...
-    },
-  ),
-];
-
-/** Annoying dune progress */
-print_endline("");
+      |> executeSideEffects;
+    let mountLog = state |> getMountLogAndReset;
 
-Alcotest.run(~argv=[|"--verbose --color"|], "Brisk", [("Core", core)]);
```
Comment (Et7f3) [inline test/Test.re]: It is conserved ? `"Brisk", [("Core", core)]`

File: test/Test.re [hunk 5]
```diff
@@ -536,286 +698,278 @@ let core = [
...
+  });
 
-      expectInt(~label="The effect should've been run", 1, effectCallCount^);
+  test("The effect should not have been run again", ({expect}) => {
+    state := state^ |> update(Components.(<Div />)) |> executeSideEffects;
 
-      expectInt(
-        ~label="The dispose should not have been run yet",
-        0,
-        effectDisposeCallCount^,
```
Comment (Et7f3) [inline test/Test.re]: Missing ?

File: test/Components.re [hunk 6]
```diff
@@ -221,7 +222,8 @@ module EmptyComponentWithAlwaysEffect = {
 };
 
 module EmptyComponentWithOnMountEffect = {
-  let%component make = (~onEffect, ~onEffectDispose, ()) => {
+  let%component make =
+                (~onEffect: unit => unit, ~onEffectDispose: unit => unit, ()) => {
```
Comment (Et7f3) [inline test/Components.re]: Why are this type annotation needed ?

Comment (ulrikstrid) [reply test/Components.re]: It complained about not being able to generalize a type, but this is only used in tests so it doesn't bleed anywhere.

File: package.json (hunk 3 above)
Comment (ulrikstrid) [reply package.json]: Correct, I removed that as well

File: test/Test.re (hunk 4 above)
Comment (ulrikstrid) [reply test/Test.re]: I could nest the `describe` some more, do you want this @wokalski ?

File: test/Test.re (hunk 5 above)
Comment (ulrikstrid) [reply test/Test.re]: I think it's on line 953, the diff is a bit hard to read here.

File: test/Test.re (hunk 4 above)
Comment (wokalski) [reply test/Test.re]: Not relevant anymore. It's a relict of the distant past.

File: test/TestHelpers.re [hunk 7]
```diff
@@ -0,0 +1,70 @@
+open TestReconciler;
+open Brisk_reconciler__Brisk_reconciler_internal;
+
+type mountElement = RenderedElement.t(node, node);
+
+type mount = list(testMountEntry);
+
+type testHostItem('a) =
+  | MountElement(mountElement): testHostItem(mount);
```
[the remark this answers is not in the document]
Comment (ulrikstrid) [reply test/TestHelpers.re]: This is not new, `TestHelpers.re` is the old `Assert.re` but I removed some stuff at the end and added a new function.

File: test/Components.re (hunk 6 above)
Comment (wokalski) [reply test/Components.re]: weird!

**Pull request merged**